containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -767,6 +810,32 @@ class UnorderedMap {
     }
   }
 
+  KOKKOS_FORCEINLINE_FUNCTION
+  bool is_op_noop() const {
+    return std::is_base_of_v<
+        insert_op_type,
+        typename UnorderedMapInsertOpTypes<
+            Kokkos::View<std::remove_const_t<std::conditional_t<
+                             std::is_void_v<Value>, int, Value>> *,
+                         Device>,
+            uint32_t,
+            std::remove_const_t<
+                std::conditional_t<std::is_void_v<Value>, int, Value>>>::NoOp>;
```

Comment:
That's pretty hard to read 🙂 
```suggestion
    using local_value_type = std::remove_const_t<std::conditional_t<std::is_void_v<Value>, int, Value>>;
    return std::is_base_of_v<
        insert_op_type,
        typename UnorderedMapInsertOpTypes<
            Kokkos::View<local_value_type *, Device>,
            uint32_t, local_value_type>::NoOp>;
```
and similarly below.